src/main/java/ch/wisv/areafiftylan/service/TicketServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package ch.wisv.areafiftylan.service;
+
+import ch.wisv.areafiftylan.exception.TicketNotFoundException;
+import ch.wisv.areafiftylan.exception.TicketNotTransferrableException;
+import ch.wisv.areafiftylan.exception.TicketUnavailableException;
+import ch.wisv.areafiftylan.model.Ticket;
+import ch.wisv.areafiftylan.model.User;
+import ch.wisv.areafiftylan.model.util.TicketType;
+import ch.wisv.areafiftylan.service.repository.TicketRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TicketServiceImpl implements TicketService {
+
+    private TicketRepository ticketRepository;
+    private UserService userService;
+
+    @Autowired
+    public TicketServiceImpl(TicketRepository ticketRepository, UserService userService) {
+        this.ticketRepository = ticketRepository;
+        this.userService = userService;
+    }
+
+    @Override
+    public Ticket getTicketByKey(String key){
+        return ticketRepository.findByKey(key).orElseThrow(() -> new TicketNotFoundException("Ticket with key " + key + " not found."));
+    }
+
+    @Override
+    public Ticket requestTicketOfType(TicketType type, User owner, boolean pickupService, boolean chMember) {
```

Comment:
Can you make this a copy of the function you removed? I'm missing the `synchronized` here, not sure if there are other changes.